src/App.tsx

```diff
@@ -14,11 +15,97 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  addProduct = (product: string) => {
+    this.setState((prevState: State): State => {
+      const { selectedGoods } = prevState;
+
+      return { selectedGoods: [...selectedGoods, product] };
+    });
+  };
```

Comment:
here you could have simply use this.state. are we required to do this bit `(prevState: State): State`?
i'm genuinely curious

Reply:
for some reason, linter wanted me to do that yesterday.
today, it works without `: State`